scripts/python/packages/parthenon_tools/parthenon_tools/phdf.py

```diff
@@ -488,24 +488,85 @@ def Get(self, variable, flatten=True, interior=False):
+
+            elif self.OutputFormatVersion == 3:
+                # Check for cell-centered data
+                if (
+                    vShape[-1] == self.MeshBlockSize[0]
+                    and vShape[-2] == self.MeshBlockSize[1]
+                    and vShape[-3] == self.MeshBlockSize[2]
+                ):
+                    fieldShape = vShape[1:-3]
+                    totalFieldEntries = np.prod(fieldShape)
+                    ndim = len(fieldShape)
+                    if ndim == 0:
+                        return self.varData[variable].ravel()
+                    else:
+                        if ndim == 1:
+                            ret = np.empty(
+                                (vShape[1], self.TotalCells),
+                                dtype=self.varData[variable].dtype,
+                            )
+                            ret[:] = np.nan
+                            for i in range(vShape[1]):
+                                ret[i] = self.varData[variable][:, i, :, :, :].ravel()
+                            assert (ret != np.nan).all()
+                            return ret
+                        elif ndim == 2:
+                            ret = np.empty(
+                                (vShape[1] * vShape[2], self.TotalCells),
+                                dtype=self.varData[variable].dtype,
+                            )
+                            ret[:] = np.nan
+                            for i in range(vShape[1]):
+                                for j in range(vShape[2]):
+                                    ret[i + vShape[1] * j] = self.varData[variable][
+                                        :, i, j, :, :, :
+                                    ].ravel()
+                            assert (ret != np.nan).all()
+                            return ret
+                        else:
+                            ret = np.empty(
+                                (vShape[1] * vShape[2] * vShape[3], self.TotalCells),
+                                dtype=self.varData[variable].dtype,
+                            )
+                            ret[:] = np.nan
+                            for i in range(vShape[1]):
+                                for j in range(vShape[2]):
+                                    for k in range(vShape[3]):
+                                        ret[
+                                            i + vShape[1] * (j + vShape[2] * k)
+                                        ] = self.varData[variable][
+                                            :, i, j, k, :, :, :
+                                        ].ravel()
+                            assert (ret != np.nan).all()
+                            return ret
+                else:
+                    # Not cell-based variable
+                    raise Exception(
+                        f"Flattening only supported for cell-based variables but requested for {variable}"
+                    )
```

Comment:
Does phdf still work for exploded views with ghost zones? I can try to check this.